tests/php/Functional/WC/Gateway/Mollie_WC_Gateway_Abstract_Test.php

```diff
@@ -33,53 +34,45 @@ public function getReturnUrl_ReturnsString_withPolylang()
         /*
          * Setup Testee
          */
-        $testee = $this
-            ->buildTesteeMock(
+        $testee = $this->buildTesteeMock(
                 Testee::class,
                 [],
                 ['getSiteUrlWithLanguage']
             )
             ->getMockForAbstractClass();
         $testee = $this->proxyFor($testee);
-        $urlFromWcApi = $this
-            ->buildTesteeMock(
-                \WooCommerce::class,
-                [],
-                ['api_request_url']
-            )
-            ->getMock();
-        $wcOrder = $this->buildTesteeMock(
-            '\\WC_Order',
-            [],
-            ['get_id', 'get_order_key']
-        )->getMock();
 
+        /*
+         * Setup Stubs
+         */
+        $urlFromWcApi = $this->createConfiguredMock(
+                \WooCommerce::class,
+                ['api_request_url'=>'http://mollie-wc.docker.myhost/wc-api/mollie_return/']
+            );
+        $wcOrder = $this->createMock('WC_Order');
+        $faker = Faker\Factory::create();
```

Comment:
I would put this into the main TestCase this way:

```
/**
     * @var Generator
     */
    protected $faker;

    /**
     * @inheritDoc
     * @throws InvalidArgumentException
     */
    protected function setUp()
    {
        parent::setUp();

        $this->setupFaker();
    }

    /**
     * Create Faker instance
     *
     * @throws InvalidArgumentException
     * @return void
     */
    protected function setupFaker(): void
    {
        $fakeFactory = new Faker\Factory();
        $this->faker = $fakeFactory->create();
    }
```

in your test then you simply need to use it.